### PR TITLE
Breadcrumb and sidenav generation gets most appropriate page version

### PR DESCRIPTION
**Pull request by kurtw**
This PR fixes two bugs brought up by @Scotchester. 
- Breadcrumb and sidenav generation used to get the page titles from the Page table in the database, leaking saved edits to `title` that were not shared or published.
- Breadcrumb and sidenav items that were previously shared but then had edits saved would return `None` as their URL. 

## Additions

- `get_appropriate_page_version` function in utils.py that grabs the page version based on revision history and request hostname.

## Changes

- `get_breadcrumbs` method on CFGOVPage and `get_secondary_nav_items` function in utils.py now use `get_appropriate_page_version`.

## Testing

- Get the latest from prod database OR share a Sublanding page with a Browse page as a child. The breadcrumb should be in the top left under the mega menu.
- **BEFORE PULLING DOWN**
 - Edit the title of this page `http://localhost:8000/admin/pages/3085/edit/` and hit `SAVE AS DRAFT`
 - Go to `http://content.localhost:8000/consumer-tools/auto-loans/plan-to-shop-for-your-auto-loan/`
 - Notice your change in the breadcrumb and that the URL ends with `None`
- Pull down the branch
 - Run `tox -- v1.tests.test_util`
 - Go back to `http://content.localhost:8000/consumer-tools/auto-loans/plan-to-shop-for-your-auto-loan/`
 - Notice your edit is not visible and the link is correct

## Review

- @rosskarchner 
- @richaagarwal 
- @kave 
- @Scotchester 

## Checklist

* [x] Changes are limited to a single goal (no scope creep)
* [x] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [x] Passes all existing automated tests
* [x] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [x] Placeholder code is flagged
* [x] Visually tested in supported browsers and devices
* [x] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### CODE REVIEW

**Comment by Scotchester** [comment]
Nice! I'll test this out tomorrow.

File: cfgov/v1/tests/test_util.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import mock
```
Comment (richaagarwal) [inline cfgov/v1/tests/test_util.py]: yayyyy tests!!! 🙌🏽

Comment (kurtw) [inline cfgov/v1/tests/test_util.py]: Bask in the glory!

**Comment by jimmynotjim** [comment]
@kurtw I'm getting a 404 when trying to navigate to the url you list. I have the latest DB dump, any ideas what I'm missing?

**Comment by jimmynotjim** [comment]
Code looks good but I couldn't duplicate the initial issue. I'll leave it to @Scotchester to confirm the fix

**Comment by Scotchester** [comment]
Tested locally and everything's working great! :+1:

**Pull request merged**